internal/controller/postgrescluster/postgres.go

```diff
@@ -628,6 +635,71 @@ func (r *Reconciler) reconcilePostgresDataVolume(
 	return pvc, err
 }
 
+// setVolumeSize compares the potential sizes from the instance spec, status
+// and limit and sets the appropriate current value.
+func (r *Reconciler) setVolumeSize(ctx context.Context, cluster *v1beta1.PostgresCluster,
+	pvc *corev1.PersistentVolumeClaim, instanceSpecName string) {
+	log := logging.FromContext(ctx)
+
+	// Capture the largest pgData volume size currently defined for a given instance set.
+	volumeRequestSize := pvc.Spec.Resources.Requests.Storage() //.Value()
+
+	// If the request value is greater than the set limit, use the limit and issue
+	// a warning event. A limit of 0 is ignorned.
+	if !pvc.Spec.Resources.Limits.Storage().IsZero() &&
+		pvc.Spec.Resources.Requests.Storage().Value() > pvc.Spec.Resources.Limits.Storage().Value() {
```

Comment:
:pick: 
```suggestion
		volumeRequestSize.Value() > pvc.Spec.Resources.Limits.Storage().Value() {
```

Reply:
See the comment above. I think I'd like to keep these values as explicitly from the pvc spec.

Reply:
Updated, see comment.